### DISCUSSION TITLE
How to specify JWT Bearer Authentication in the OpenAPI spec?

### DISCUSSION THREAD

**Discussion by McMerph** [RobinTail/express-zod-api#803] [Q&A] [answered]
I want to specify that a particular middleware uses JWT Bearer Authentication.
So, I added:
```
security: { type: "bearer", format: "JWT" }
```
to `createMiddleware`.
Unfortunately it didn't work -(
Error is:
```
TypeError: methods[security.type] is not a function
    at {{PROJECT_DIR}}/node_modules/express-zod-api/src/open-api-helpers.ts:801:69
    at {{PROJECT_DIR}}/node_modules/express-zod-api/src/logical-container.ts:40:15
    at Array.map (<anonymous>)
    at mapLogicalContainer ({{PROJECT_DIR}}/node_modules/express-zod-api/src/logical-container.ts:37:28)
    at depictSecurity ({{PROJECT_DIR}}/node_modules/express-zod-api/src/open-api-helpers.ts:800:29)
    at onEndpoint ({{PROJECT_DIR}}/node_modules/express-zod-api/src/open-api.ts:120:25)
    at {{PROJECT_DIR}}/node_modules/express-zod-api/src/routing-walker.ts:46:9
    at Array.forEach (<anonymous>)
    at {{PROJECT_DIR}}/node_modules/express-zod-api/src/routing-walker.ts:45:15
    at Array.forEach (<anonymous>)
```
What am I doing wrong?

---
```
security: { type: "bearer" }
```
doesn't work either.

Actually
```
security: { or: [{ type: "bearer", format: "JWT" }] }
```
or
```
security: { and: [{ type: "bearer", format: "JWT" }] }
```
both work but https://editor.swagger.io/ reports structural errors in this case (screenshot provided).
Both the middleware and the endpoint have an empty input (`input: z.object({})`) if that matters.
<img width="1725" alt="image" src="https://user-images.githubusercontent.com/29835709/217489320-92c95070-6739-49a9-8c65-3f3a6308f800.png">

**Comment by RobinTail** [comment]
This actually looks like a bug.
Let me take some time to figure this out...
And thank you for the PR that fixes it. I will take it into account, @McMerph

**Comment by RobinTail** [comment]
All currently existing tests actually do not cover the case of a singular `Security`. 🤦🏽‍♀️ 
Such a good finding, @McMerph

**Comment by RobinTail** [comment]
Confirming the bug.

**Comment by RobinTail** [comment, marked as answer]
Fixed in v8.9.2 🚀
